cmd/authgear/cmd/cmdsearch/reindex.go

```diff
@@ -0,0 +1,96 @@
+package cmdsearch
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+
+	authgearcmd "github.com/authgear/authgear-server/cmd/authgear/cmd"
+	cmdpgsearch "github.com/authgear/authgear-server/cmd/authgear/pgsearch"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/infra/db"
+	"github.com/authgear/authgear-server/pkg/lib/infra/db/searchdb"
+	"github.com/authgear/authgear-server/pkg/lib/search/pgsearch"
+	authgearlog "github.com/authgear/authgear-server/pkg/util/log"
+)
+
+var cmdSearchReindex = &cobra.Command{
+	Use:   "reindex { app-id }",
+	Short: "Reindex all documents of a given app into the search index",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return fmt.Errorf("expected exactly 1 argument of app ID")
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		binder := authgearcmd.GetBinder()
+
+		dbURL, err := binder.GetRequiredString(cmd, authgearcmd.ArgDatabaseURL)
+		if err != nil {
+			return err
+		}
+
+		dbSchema, err := binder.GetRequiredString(cmd, authgearcmd.ArgDatabaseSchema)
+		if err != nil {
+			return err
+		}
+
+		searchDBURL, err := binder.GetRequiredString(cmd, authgearcmd.ArgSearchDatabaseURL)
+		if err != nil {
+			return err
+		}
+
+		searchDBSchema, err := binder.GetRequiredString(cmd, authgearcmd.ArgSearchDatabaseSchema)
+		if err != nil {
+			return err
+		}
+
+		dbCredentials := &cmdpgsearch.CmdDBCredential{
+			DatabaseURL:    dbURL,
+			DatabaseSchema: dbSchema,
+		}
+
+		searchDatabaseCredentials := &config.SearchDatabaseCredentials{
+			DatabaseURL:    searchDBURL,
+			DatabaseSchema: searchDBSchema,
+		}
+
+		dbPool := db.NewPool()
+		loggerFactory := authgearlog.NewFactory(authgearlog.LevelInfo)
+
+		reindexApp := func(appID string) error {
+			ctx := context.Background()
+			log.Printf("App (%s): reindexing\n", appID)
+			searchdbHandle := searchdb.NewHandle(
+				dbPool,
+				config.NewDefaultDatabaseEnvironmentConfig(),
+				searchDatabaseCredentials,
+				loggerFactory)
+			searchdbSQLBuilder := searchdb.NewSQLBuilder(searchDatabaseCredentials)
+			searchdbSQLExecutor := searchdb.NewSQLExecutor(searchdbHandle)
+			store := pgsearch.NewStore(config.AppID(appID), searchdbSQLBuilder, searchdbSQLExecutor)
+
+			reindexer := cmdpgsearch.NewReindexer(dbPool, dbCredentials, cmdpgsearch.CmdAppID(appID))
+			err = searchdbHandle.WithTx(ctx, func(ctx context.Context) error {
```

Comment:
If the app contains many users, then this transaction could be opened for a long time. Maybe it is better to let the reindexer to manage the transaction. For now, we can just do 1 txn for 1 source (or user).